internal/engine/eval/vulncheck/review.go

```diff
@@ -312,30 +275,64 @@ func (ra *reviewPrHandler) findPreviousReview(ctx context.Context) error {
 	}
 
 	ra.minderReview = nil
-	for _, r := range reviews {
-		if strings.HasPrefix(r.GetBody(), reviewBodyMagicComment) && r.GetState() != "DISMISSED" {
-			ra.minderReview = r
+	for i := len(reviews) - 1; i >= 0; i-- {
+		review := reviews[i]
+		isMinder := review.GetUser().GetLogin() == stacklokBotUser && strings.HasPrefix(review.GetBody(), reviewBodyMagicComment)
+		if isMinder && review.GetState() != "DISMISS" {
+			ra.minderReview = review
+			break
+		}
+
+	}
+
+	return nil
+}
+
+func (ra *reviewPrHandler) findPreviousStatusComment(ctx context.Context) error {
+	comments, err := ra.cli.ListComments(ctx, ra.pr.RepoOwner, ra.pr.RepoName, int(ra.pr.Number),
```

Comment:
oh so `ListComments` allows you to set the sorting, but `ListReviews` does not?

Reply:
Unfortunately this part of the GitHub API is a bit inconsistent :octocat: 
See https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#list-reviews-for-a-pull-request versus https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28#list-review-comments-in-a-repository

Reply:
Ah, I think this is the crux of the issues I was seeing. So the ` ra.cli.ListComments` method wraps `c.client.PullRequests.ListComments`, its result is what the code iterates over and saves into `ra.minderStatusReport`. But, the comments are created with `ra.cli.CreateComment` which uses `c.client.Issues.CreateComment` - it's a different API. The `Issues` one is for what's usually referred to as comments, but the `PullRequest` API is for iterating over review comments, those that are submitted either inline or as the review summary.

I know this work has been going on for some time, I wonder if there's a reason those two got mixed up?